### PR TITLE
Chore/issues/47 add non linear sklearn regressor adapter hist gbdt and wire into model router

### LINKED ISSUE #47: Add non-linear sklearn regressor adapter (`hist_gbdt`) and wire into model router
- **Problem:** Current model set is baselines + linear Ridge; we need a non-linear model for better performance and more meaningful comparisons.
- **Scope:**
  - Add a new sklearn adapter using a non-linear regressor and expose it under model id: **`hist_gbdt`**.
  - Default proposal: `sklearn.ensemble.HistGradientBoostingRegressor`.
  - Implement under `src/finsight/infrastructure/ml/sklearn/` and register in `SklearnModelRouter`.
  - Ensure evaluation follows existing conventions:
    - Build arrays via `train_df.loc[:, feature_columns].to_numpy(dtype=float)` (order matters)
    - Fit model (set `random_state` for deterministic behavior)
    - Predict on test set
    - Return `ModelEvaluationResult` with:
      - `trained_artifact` (the fitted model / pipeline)
      - `predictions` (same structure as other adapters)
      - `metrics` (existing + new metrics if Issue #1 is implemented)
  - Update container wiring (`src/finsight/bootstrap/container.py`) to include the adapter.
  - Update `config/config.yaml` model catalog to include:
    - id: `hist_gbdt`
    - label, supports_training/supports_prediction flags
- **Acceptance Criteria:**
  - `hist_gbdt` is selectable for training (via CLI `--model-types hist_gbdt`) and produces run artifacts.
  - Metrics + predictions are saved for `hist_gbdt`.
  - Unit tests verify routing and deterministic evaluation (fixed `random_state`).
  - No changes required to application/domain layers beyond model selection and wiring.

### PR DESCRIPTION
**Pull request by KTomy7**
This pull request adds support for a new machine learning model, Histogram Gradient Boosting (`hist_gbdt`), to the FinSight application. The changes include implementing the model adapter, updating the dependency injection container, expanding the CLI and documentation to reference the new model, and updating the architecture overview.

**New Model Support**

* Added a new adapter class `HistGradientBoostingModel` in `ml/sklearn/tree.py` to provide training and evaluation for the `hist_gbdt` model using scikit-learn's `HistGradientBoostingRegressor`. This includes feature selection, hyperparameter configuration, metrics computation, and feature importance extraction.
* Registered `HistGradientBoostingModel` in the dependency injection container (`container.py`) and made it available to the model router alongside existing models.

**Configuration and Documentation Updates**

* Added `hist_gbdt` to the list of supported models in `config/config.yaml`, enabling it for training and prediction.
* Updated CLI documentation to include `hist_gbdt` as a supported model for training, comparison, and forecasting, with revised examples and outputs. 
* Revised the architecture overview to document the new model adapter and its mapping to the `ModelPort` interface.

### CODE REVIEW

File: src/finsight/infrastructure/ml/sklearn/tree.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+            min_samples_leaf=min_samples_leaf,
+            l2_regularization=l2_regularization,
+        )
+        model.fit(x_train, y_train)
+        y_pred = model.predict(x_test)
+        hist_gbdt_model = model
+
+        # Compute feature importances (if available)
+        try:
+            importances = getattr(hist_gbdt_model, "feature_importances_", None)
```
[the remark this answers is not in the document]
Comment (KTomy7) [reply src/finsight/infrastructure/ml/sklearn/tree.py:73]: Fixed it

File: tests/integration/test_hist_gbdt_e2e.py [hunk 2]
```diff
@@ -0,0 +1,312 @@
+"""End-to-end integration tests for HistGradientBoostingModel (hist_gbdt)."""
+from __future__ import annotations
+
+import json
+from datetime import date
```
[the remark this answers is not in the document]
Comment (KTomy7) [reply tests/integration/test_hist_gbdt_e2e.py:5]: Fixed it
